### DISCUSSION TITLE
reviving the old pylon library?!

### DISCUSSION THREAD

**Discussion by chiyu1203** [bonsai-rx/bonsai#1395] [Ideas]
Dear Bonsai team
I am interested in knowing whether there is a possibility to revive the old pylon library, which requires pylon runtime 5.1.0, or to downgrade the existing pylon library so that it would use pylon runtime 5.1.0.
The reason why I wanted to use pylon 5.1.0 is because I want to use an open-source tracking software [fictrac](https://github.com/rjdmoore/fictrac), which can be soothingly compiled with pylon 5.1.0, (needs additional hack for pylon 6.x.x and no information about pylon 7.1.0 yet), for one video recording while filming other parts of the animal with other cameras on Bonsai. I also learned that it is possible to install different pylon version in the same PC. Here is the [reference](https://zh.docs.baslerweb.com/knowledge/running-different-pylon-versions-on-a-computer). However, I still have no idea to manage each runtime how to get bonsai to use 7.1.0 and the fictrac to use 5.1.0.......
Anyway, just some ideas. if this is too complicated, I can also install different pylon in different PC.
Best
Chiyu

**Comment by bruno-f-cruz** [comment]
As explained [here](https://bonsai-rx.org/docs/articles/environments.html) you can manually edit the `bonsai.config` file to specify package versions. Just be careful to make sure that the environment dependencies are able to resolve. If you install two versions, Bonsai should be able pick the ones it needs for the currently installed package if I recall correctly. Have you actually tried this?

**Reply by chiyu1203** [reply]
Hi!! Thanks for pointing out that! It was really helpful! In the beginning, I cant find the old pylon library (0.1.0) on the package manager, but luckily I found an old bonsai that had installed  the old package. I moved that package and changed the config and it seems to be working now. I am very happy about it!

**Reply by the-dude-man** [reply]
Hi Chiyu1203 - my name is David Deutsch. I recently opened a new Drosophila lab in Israel (https://deutschlab.haifa.ac.il/ ) and we are considering to use fictrac with Bonsai in some setups. Seems like you are do doing the same thing, so I like to ask for you experience. If OK with you  - can you please send me a Hi email to ddeutsch1@univ.haifa.ac.il? Thanks!!
